### DISCUSSION TITLE
Dimensionally-stacked heatmaps

### DISCUSSION THREAD

**Discussion by EwoutH** [mwaskom/seaborn#3791] [Q&A]
I recently implemented a custom extension to Seaborn's `heatmap` function to visualize hierarchical data with multi-level indices. This approach reshapes and formats hierarchical data into a dimensionally-stacked heatmap, enabling clear representation of multi-dimensional relationships (e.g., parameter sweeps or policy comparisons).

Here’s an example visualization:  
![heatmap_mean_av_speed](https://github.com/user-attachments/assets/51a043a4-b2e9-47bf-ac4c-c650924bc2bf)

Note the custom formatting of axis ticks to display hierarchical levels and dividing lines to visually separate higher-level categories (these could be thicker or a color though).

This type of plot is particularly useful for multi-dimensional data analysis, like parameter sweeps or experimental results. Practically anything you want to visualize with more than two dimensions. It was originally [inspired](https://emaworkbench.readthedocs.io/en/latest/indepth_tutorial/open-exploration.html#dimensional-stacking) from the EMA workbench.

@mwaskom Would you find such functionality to be useful for Seaborn, or should it remain an external utility built on top of it? If it’s a potential fit, I’d be happy to refine the design and open an PR.

Thanks for maintaining such an excellent library!

**Comment by simon-hirsch** [comment]
Super cool, would you like to share some of the if this does not make it into seaborn?

**Reply by EwoutH** [reply]
You can find some here:
- Code: https://github.com/EwoutH/urban-self-driving-effects/blob/main/model/analysis_ff.ipynb
- Images: https://github.com/EwoutH/urban-self-driving-effects/tree/main/img/exp4

**Reply by simon-hirsch** [reply]
Thanks! :)
